### ISSUE TITLE
`eb config put` fails for PHP platform

### ISSUE THREAD

**Issue by simplesteph** [aws/aws-elastic-beanstalk-cli#32]
### Description

fatal error with the `eb config put`

```
eb config put prod
ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "PHP 7.4 running on 64bit Amazon Linux 2".
```

### Steps to reproduce

```
eb init
> choose PHP 7.4 as a platform
echo "Hello World" > index.html
eb create dev-env
eb open
eb status
eb config save prod --cfg initial-configuration
eb config put prod
ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "PHP 7.4 running on 64bit Amazon Linux 2".
```

### Observed result

```
2020-05-15 17:38:11,958 (DEBUG) cement.core.foundation : laying cement for the 'eb' application
2020-05-15 17:38:11,958 (DEBUG) cement.core.hook : defining hook 'pre_setup'
2020-05-15 17:38:11,958 (DEBUG) cement.core.hook : defining hook 'post_setup'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'pre_run'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'post_run'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'pre_argument_parsing'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'post_argument_parsing'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'pre_close'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'post_close'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'signal'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'pre_render'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : defining hook 'post_render'
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : registering hook 'add_handler_override_options' from cement.core.foundation into hooks['post_setup']
2020-05-15 17:38:11,959 (DEBUG) cement.core.hook : registering hook 'handler_override' from cement.core.foundation into hooks['post_argument_parsing']
2020-05-15 17:38:11,959 (DEBUG) cement.core.handler : defining handler type 'extension' (IExtension)
2020-05-15 17:38:11,959 (DEBUG) cement.core.handler : defining handler type 'log' (ILog)
2020-05-15 17:38:11,959 (DEBUG) cement.core.handler : defining handler type 'config' (IConfig)
2020-05-15 17:38:11,959 (DEBUG) cement.core.handler : defining handler type 'mail' (IMail)
2020-05-15 17:38:11,959 (DEBUG) cement.core.handler : defining handler type 'plugin' (IPlugin)
2020-05-15 17:38:11,960 (DEBUG) cement.core.handler : defining handler type 'output' (IOutput)
2020-05-15 17:38:11,960 (DEBUG) cement.core.handler : defining handler type 'argument' (IArgument)
2020-05-15 17:38:11,960 (DEBUG) cement.core.handler : defining handler type 'controller' (IController)
2020-05-15 17:38:11,960 (DEBUG) cement.core.handler : defining handler type 'cache' (ICache)
2020-05-15 17:38:11,960 (DEBUG) cement.core.handler : registering handler '<class 'cement.core.extension.CementExtensionHandler'>' into handlers['extension']['cement']
2020-05-15 17:38:11,989 (DEBUG) cement.ext.ext_plugin : plugin config dir /etc/eb/plugins.d does not exist.
2020-05-15 17:38:11,989 (DEBUG) cement.ext.ext_plugin : plugin config dir /Users/stephanemaarek/.eb/plugins.d does not exist.
2020-05-15 17:38:11,997 (DEBUG) ebcli.core.hooks : -- EBCLI Version: 3.18.1
2020-05-15 17:38:11,997 (DEBUG) ebcli.core.hooks : -- Python Version: 3.7.7 (default, Mar 10 2020, 15:43:33)
[Clang 11.0.0 (clang-1100.0.33.17)]
2020-05-15 17:38:11,997 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,000 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,003 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,006 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,008 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,011 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,014 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,017 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,020 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,022 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,025 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/stephanemaarek/junk/eb/HelloWorld/HelloWorld
2020-05-15 17:38:12,027 (DEBUG) ebcli.lib.elasticbeanstalk : Inside get_available_solution_stacks api wrapper
2020-05-15 17:38:12,028 (DEBUG) ebcli.lib.aws : Creating new Botocore Session
2020-05-15 17:38:12,028 (DEBUG) ebcli.lib.aws : Botocore version: 1.15.47
2020-05-15 17:38:12,039 (DEBUG) ebcli.lib.aws : Creating new Botocore Client for elasticbeanstalk
2020-05-15 17:38:12,127 (DEBUG) ebcli.lib.aws : Successfully created session for elasticbeanstalk
2020-05-15 17:38:12,128 (DEBUG) ebcli.lib.aws : Making api call: (elasticbeanstalk, list_available_solution_stacks) to region: eu-west-1 with args:{}
2020-05-15 17:38:12,625 (DEBUG) ebcli.lib.aws : API call finished, status = 200
2020-05-15 17:38:12,625 (DEBUG) ebcli.lib.aws : Response: {'SolutionStacks': ['64bit Amazon Linux 2018.03 v4.14.3 running Node.js', '64bit Amazon Linux 2018.03 v4.10.1 running Node.js', '64bit Amazon Linux 2018.03 v2.9.6 running PHP 5.4', '64bit Amazon Linux 2018.03 v2.9.6 running PHP 5.5', '64bit Amazon Linux 2018.03 v2.9.6 running PHP 5.6', '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.0', '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.1', '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.2', '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.3', '64bit Amazon Linux 2018.03 v2.8.14 running PHP 7.2', '64bit Amazon Linux 2018.03 v2.9.10 running Python 3.6', '64bit Amazon Linux 2018.03 v2.9.10 running Python 3.4', '64bit Amazon Linux 2018.03 v2.9.10 running Python', '64bit Amazon Linux 2018.03 v2.9.10 running Python 2.7', '64bit Amazon Linux 2018.03 v2.9.1 running Python 3.6', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.6 (Puma)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.5 (Puma)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.4 (Puma)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.3 (Puma)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.2 (Puma)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.1 (Puma)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.0 (Puma)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.6 (Passenger Standalone)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.5 (Passenger Standalone)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.4 (Passenger Standalone)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.3 (Passenger Standalone)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.2 (Passenger Standalone)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.1 (Passenger Standalone)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.0 (Passenger Standalone)', '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 1.9.3', '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 8.5 Java 8', '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 8 Java 8', '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 7 Java 7', '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 7 Java 6', '64bit Windows Server Core 2019 v2.5.5 running IIS 10.0', '64bit Windows Server 2019 v2.5.5 running IIS 10.0', '64bit Windows Server Core 2016 v2.5.5 running IIS 10.0', '64bit Windows Server 2016 v2.5.5 running IIS 10.0', '64bit Windows Server Core 2012 R2 v2.5.5 running IIS 8.5', '64bit Windows Server 2012 R2 v2.5.5 running IIS 8.5', '64bit Windows Server Core 2016 v1.2.0 running IIS 10.0', '64bit Windows Server 2016 v1.2.0 running IIS 10.0', '64bit Windows Server Core 2012 R2 v1.2.0 running IIS 8.5', '64bit Windows Server 2012 R2 v1.2.0 running IIS 8.5', '64bit Windows Server 2012 v1.2.0 running IIS 8', '64bit Windows Server Core 2012 R2 running IIS 8.5', '64bit Windows Server 2012 R2 running IIS 8.5', '64bit Windows Server 2012 running IIS 8', '64bit Amazon Linux 2018.03 v2.15.0 running Docker 19.03.6-ce', '64bit Amazon Linux 2018.03 v2.20.2 running Multi-container Docker 19.03.6-ce (Generic)', '64bit Amazon Linux 2018.03 v2.15.2 running Multi-container Docker 18.06.1-ce (Generic)', '64bit Debian jessie v2.15.0 running Go 1.4 (Preconfigured - Docker)', '64bit Debian jessie v2.15.0 running Go 1.3 (Preconfigured - Docker)', '64bit Debian jessie v2.15.0 running Python 3.4 (Preconfigured - Docker)', '64bit Amazon Linux 2 v3.0.1 running Corretto 11', '64bit Amazon Linux 2 v3.0.1 running Corretto 8', '64bit Amazon Linux 2 v3.0.1 running Ruby 2.7', '64bit Amazon Linux 2 v3.0.1 running Ruby 2.6', '64bit Amazon Linux 2 v3.0.1 running Ruby 2.5', '64bit Amazon Linux 2 v5.0.1 running Node.js 12', '64bit Amazon Linux 2 v5.0.1 running Node.js 10', '64bit Amazon Linux 2 v3.0.1 running Go 1', '64bit Amazon Linux 2 v3.0.1 running Docker', '64bit Amazon Linux 2 v3.0.1 running PHP 7.4', '64bit Amazon Linux 2 v3.0.1 running PHP 7.3', '64bit Amazon Linux 2 v3.0.1 running PHP 7.2', '64bit Amazon Linux 2 v3.0.1 running Python 3.7', '64bit Amazon Linux 2018.03 v2.10.7 running Java 8', '64bit Amazon Linux 2018.03 v2.10.7 running Java 7', '64bit Amazon Linux 2018.03 v2.15.3 running Go 1.14.2', '64bit Amazon Linux 2018.03 v2.6.22 running Packer 1.0.3', '64bit Amazon Linux 2018.03 v2.15.0 running GlassFish 5.0 Java 8 (Preconfigured - Docker)'], 'SolutionStackDetails': [{'SolutionStackName': '64bit Amazon Linux 2018.03 v4.14.3 running Node.js', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v4.10.1 running Node.js', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.6 running PHP 5.4', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.6 running PHP 5.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.6 running PHP 5.6', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.0', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.1', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.2', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.6 running PHP 7.3', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.8.14 running PHP 7.2', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.10 running Python 3.6', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.10 running Python 3.4', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.10 running Python', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.10 running Python 2.7', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.9.1 running Python 3.6', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.6 (Puma)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.5 (Puma)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.4 (Puma)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.3 (Puma)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.2 (Puma)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.1 (Puma)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.0 (Puma)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.6 (Passenger Standalone)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.5 (Passenger Standalone)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.4 (Passenger Standalone)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.3 (Passenger Standalone)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.2 (Passenger Standalone)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.1 (Passenger Standalone)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 2.0 (Passenger Standalone)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.11.6 running Ruby 1.9.3', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 8.5 Java 8', 'PermittedFileTypes': ['war', 'zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 8 Java 8', 'PermittedFileTypes': ['war', 'zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 7 Java 7', 'PermittedFileTypes': ['war', 'zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v3.3.6 running Tomcat 7 Java 6', 'PermittedFileTypes': ['war', 'zip']}, {'SolutionStackName': '64bit Windows Server Core 2019 v2.5.5 running IIS 10.0', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2019 v2.5.5 running IIS 10.0', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server Core 2016 v2.5.5 running IIS 10.0', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2016 v2.5.5 running IIS 10.0', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server Core 2012 R2 v2.5.5 running IIS 8.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2012 R2 v2.5.5 running IIS 8.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server Core 2016 v1.2.0 running IIS 10.0', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2016 v1.2.0 running IIS 10.0', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server Core 2012 R2 v1.2.0 running IIS 8.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2012 R2 v1.2.0 running IIS 8.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2012 v1.2.0 running IIS 8', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server Core 2012 R2 running IIS 8.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2012 R2 running IIS 8.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Windows Server 2012 running IIS 8', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.15.0 running Docker 19.03.6-ce', 'PermittedFileTypes': []}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.20.2 running Multi-container Docker 19.03.6-ce (Generic)', 'PermittedFileTypes': ['zip', 'json']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.15.2 running Multi-container Docker 18.06.1-ce (Generic)', 'PermittedFileTypes': ['zip', 'json']}, {'SolutionStackName': '64bit Debian jessie v2.15.0 running Go 1.4 (Preconfigured - Docker)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Debian jessie v2.15.0 running Go 1.3 (Preconfigured - Docker)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Debian jessie v2.15.0 running Python 3.4 (Preconfigured - Docker)', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Corretto 11', 'PermittedFileTypes': ['jar', 'zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Corretto 8', 'PermittedFileTypes': ['jar', 'zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Ruby 2.7', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Ruby 2.6', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Ruby 2.5', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v5.0.1 running Node.js 12', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v5.0.1 running Node.js 10', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Go 1', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Docker', 'PermittedFileTypes': []}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running PHP 7.4', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running PHP 7.3', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running PHP 7.2', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2 v3.0.1 running Python 3.7', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.10.7 running Java 8', 'PermittedFileTypes': ['jar', 'zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.10.7 running Java 7', 'PermittedFileTypes': ['jar', 'zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.15.3 running Go 1.14.2', 'PermittedFileTypes': ['zip']}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.6.22 running Packer 1.0.3', 'PermittedFileTypes': []}, {'SolutionStackName': '64bit Amazon Linux 2018.03 v2.15.0 running GlassFish 5.0 Java 8 (Preconfigured - Docker)', 'PermittedFileTypes': ['zip']}], 'ResponseMetadata': {'RequestId': '8c1c568b-9b16-40ef-ac8e-39e976b2458a', 'HTTPStatusCode': 200, 'date': 'Fri, 15 May 2020 16:38:12 GMT', 'RetryAttempts': 0}}
2020-05-15 17:38:12,626 (DEBUG) ebcli.lib.elasticbeanstalk : Solution Stack result size = 72
2020-05-15 17:38:12,641 (DEBUG) ebcli.lib.elasticbeanstalk : Inside list_platform_versions api wrapper
2020-05-15 17:38:12,641 (DEBUG) ebcli.lib.aws : Making api call: (elasticbeanstalk, list_platform_versions) to region: eu-west-1 with args:{'Filters': [{'Type': 'PlatformOwner', 'Operator': '=', 'Values': ['self']}]}
2020-05-15 17:38:12,734 (DEBUG) ebcli.lib.aws : API call finished, status = 200
2020-05-15 17:38:12,735 (DEBUG) ebcli.lib.aws : Response: {'PlatformSummaryList': [], 'ResponseMetadata': {'RequestId': 'ff6f5635-3638-4be7-a4d7-c132570e9c29', 'HTTPStatusCode': 200, 'date': 'Fri, 15 May 2020 16:38:11 GMT', 'RetryAttempts': 0}}
2020-05-15 17:38:12,742 (INFO) eb : Traceback (most recent call last):
  File "/Users/stephanemaarek/Library/Python/3.7/lib/python/site-packages/ebcli/core/ebrun.py", line 62, in run_app
    app.run()
  File "/Users/stephanemaarek/Library/Python/3.7/lib/python/site-packages/cement/core/foundation.py", line 797, in run
    return_val = self.controller._dispatch()
  File "/Users/stephanemaarek/Library/Python/3.7/lib/python/site-packages/cement/core/controller.py", line 472, in _dispatch
    return func()
  File "/Users/stephanemaarek/Library/Python/3.7/lib/python/site-packages/cement/core/controller.py", line 478, in _dispatch
    return func()
  File "/Users/stephanemaarek/Library/Python/3.7/lib/python/site-packages/ebcli/controllers/config.py", line 93, in put
    platform = solution_stack_ops.find_solution_stack_from_string(platform)
  File "/Users/stephanemaarek/Library/Python/3.7/lib/python/site-packages/ebcli/operations/solution_stack_ops.py", line 103, in find_solution_stack_from_string
    raise NotFoundError(alerts['platform.invalidstring'].format(solution_string))
ebcli.objects.exceptions.NotFoundError: Elastic Beanstalk can't find a platform version that matches "PHP 7.4 running on 64bit Amazon Linux 2".
```

### Expected result

The command to complete

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

```
$ eb --version
EB CLI 3.18.1 (Python 3.7.7)
$ aws --version
aws-cli/2.0.10 Python/3.7.4 Darwin/19.4.0 botocore/2.0.0dev14
```

# Important notes:

In my `prod.cfg.yml`:
```
Platform:
  PlatformArn: arn:aws:elasticbeanstalk:eu-west-1::platform/PHP 7.4 running on 64bit Amazon Linux 2/3.0.1
```

In the command output: 
```
Elastic Beanstalk can't find a platform version that matches "PHP 7.4 running on 64bit Amazon Linux 2
```

I think the issue is due to some parsing... it seems the `3.0.1` is missing

**Comment by halcarleton** [comment]
Hi @simplesteph, thanks for reporting this. I have identified the bug and we will work on getting a fix implemented. In the meantime here is a workaround for you:

```
$ eb init --platform "arn:aws:elasticbeanstalk:eu-west-1::platform/PHP 7.4 running on 64bit Amazon Linux 2/3.0.1"
$ eb config put initial-configuration
```

This will update `default_platform` in your `.elasticbeanstalk/config.yml` file to that specific platform version arn. The bug is in the platform lookup logic around that value, so if you explicitly provide an arn it can skip that portion of the logic.

**Comment by crocLeMonde** [comment]
Thanks @halcarleton for your quick answer.

This problem seems to be solved this morning without doing anything on my side, it seems there was a quick change or rollback on AWS side. I may be wrong but I have no other explanation.

**Comment by AbderrahmenISSA** [comment]
The command  :
`$ eb init --platform "arn:aws:elasticbeanstalk:eu-west-1::platform/PHP 7.4 running on 64bit Amazon Linux 2/3.0.1"`
didn't pass. I have the following error:
 `ERROR: InvalidParameterValueError - Platform 'arn:aws:elasticbeanstalk:eu-west-1::platform/PHP 7.4 running on 64bit Amazon Linux 2/3.0.1' does not exist.`
So I manually changed the file `.elasticbeanstalk/config.yml` with the arn like this :
` default_platform: arn:aws:elasticbeanstalk:eu-west-1::platform/PHP 7.4 running on 64bit Amazon Linux 2/3.0.1` and it's OK for now.

**Comment by chiantansaha** [comment]
This issue still persists, I am able to resolve it with the workaround suggested by @halcarleton

a483e78392af:HelloWorld $ eb config put prod
ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "PHP 7.4 running on 64bit Amazon Linux 2".


a483e78392af:HelloWorld $ eb init --platform "arn:aws:elasticbeanstalk:eu-west-1::platform/PHP 7.4 running on 64bit Amazon Linux 2/3.1.1"

a483e78392af:HelloWorld  $ eb config put prod
a483e78392af:HelloWorld $ eb config dev-env --cfg prod
Printing Status:
2020-10-03 09:16:57    INFO    Environment update is starting.      
2020-10-03 09:17:04    INFO    Updating environment dev-env's configuration settings.
2020-10-03 09:17:55    INFO    Successfully deployed new configuration to environment.

**Comment by lesterpotter** [comment]
Experiencing same error using awsebcli 3.19.1 with a different region (us-east-1).  As time goes by, the work around requires a different platform version ARN (3.1.2).

To find the correct platform name of for the ARN workaround, look at config.yml for the platform (and adjust your ARN region, too).

**Comment by JohnLockwood** [comment]
@simplesteph Thanks for the report and @halcarleton thank you for the workaround.

**Comment by memark** [comment]
@halcarleton This bug, reported in May last year, is still present. The work-around you suggested is not working. Is there a fix coming..?

**Comment by 65** [comment]
I experienced this issue today also. EB CLI 3.19.3 (Python 3.9.1) 

ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "PHP 7.3 running on 64bit Amazon Linux"

arn:aws:elasticbeanstalk:ap-southeast-2::platform/PHP 7.3 running on 64bit Amazon Linux/2.9.14 

Adding --platform to the eb config put results in an unrecognized argument error

eb config put staging-sc6 --platform "arn:aws:elasticbeanstalk:ap-southeast-2::platform/PHP 7.3 running on 64bit Amazon Linux/2.9.14"
usage: eb config < |save|get|put|list|delete> <name> [options ...]
eb: error: unrecognized arguments: --platform arn:aws:elasticbeanstalk:ap-southeast-2::platform/PHP 7.3 running on 64bit Amazon Linux/2.9.14

**Comment by DarwinJS** [comment]
Same exact problem on eb cli 3.19.3 - is a fix forthcoming?

**Comment by Palakpatel67** [comment]
This should be fixed in [awsebcli version 3.19.4](https://pypi.org/project/awsebcli/3.19.4/).

**Comment by purdy** [comment]
I'm running into this issue where my config .yml looks like this:

```yml
Platform:
  PlatformArn: arn:aws:elasticbeanstalk:us-east-1::platform/PHP 8.2 running on 64bit
    Amazon Linux 2023/4.0.1
```

And when I run `eb platform show`, it doesn't get the full platform name:

```
% eb platform show
ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "PHP 8.1 running on 64bit Amazon Linux 2".
```
